src/main/webapp/create-cache/create-cache.controller.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+angular.module('managementConsole')
+  .controller('createCacheCtrl', [
+    '$scope',
+    '$state',
+    '$stateParams',
+    'utils',
+    'modelController',
+    'cacheCreateController',
+    function ($scope, $state, $stateParams, utils, modelController, cacheCreateController) {
+      //-- Variables --//
+
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      if (!$stateParams.clusterName) {
+        $state.go('error404');
+      }
+
+      var server = modelController.getServer();
+      var clusters = server.getClusters();
+      $scope.currentCluster = server.getCluster(clusters, $stateParams.clusterName);
+      $scope.newlyCreatedCache = {};
+      $scope.selectedCacheType = 'distributed-cache';
+      $scope.isCreateEnabled = false;
+
+      $scope.createReplicatedOrDistributedCache = function (cache, address, cacheType) {
```

Comment:
This is a large function with cache-specific logic, so I would suggest moving it to some service that is related to cache or maybe a new service if non of existing services is good enough. It seems to me like there is some scope-related logic at the end of the function, but we can pass that as a callback to this function.